Common/Securities/Option/OptionFilterUniverse.cs

```diff
@@ -224,6 +225,483 @@ public OptionFilterUniverse PutsOnly()
+        }
+
+        /// <summary>
+        /// Sets universe of 2 call contracts with the same strike price and different expiration dates, with closest match to the criteria given
+        /// </summary>
+        /// <param name="strikeFromAtm">The desire strike price distance from the current underlying price</param>
+        /// <param name="nearDaysTillExpiry">The desire days till expiry of the closer contract from the current time</param>
+        /// <param name="farDaysTillExpiry">The desire days till expiry of the further conrtact from the current time</param>
+        /// <remarks>Applicable to Long and Short Call Calendar Spread Option Strategy</remarks>
+        /// <returns>Universe with filter applied</returns>
+        public OptionFilterUniverse CallCalendarSpread(decimal strikeFromAtm = 0, int nearDaysTillExpiry = 30, int farDaysTillExpiry = 60)
+        {
+            return CalendarSpread(OptionRight.Call, strikeFromAtm, nearDaysTillExpiry, farDaysTillExpiry);
+        }
+
+        /// <summary>
+        /// Sets universe of 2 put contracts with the same strike price and different expiration dates, with closest match to the criteria given
+        /// </summary>
+        /// <param name="strikeFromAtm">The desire strike price distance from the current underlying price</param>
+        /// <param name="nearDaysTillExpiry">The desire days till expiry of the closer contract from the current time</param>
+        /// <param name="farDaysTillExpiry">The desire days till expiry of the further conrtact from the current time</param>
+        /// <remarks>Applicable to Long and Short Put Calendar Spread Option Strategy</remarks>
+        /// <returns>Universe with filter applied</returns>
+        public OptionFilterUniverse PutCalendarSpread(decimal strikeFromAtm = 0, int nearDaysTillExpiry = 30, int farDaysTillExpiry = 60)
+        {
+            return CalendarSpread(OptionRight.Put, strikeFromAtm, nearDaysTillExpiry, farDaysTillExpiry);
+        }
+
+        private OptionFilterUniverse CalendarSpread(OptionRight right, decimal strikeFromAtm, int nearDaysTillExpiry, int farDaysTillExpiry)
+        {
+            if (farDaysTillExpiry <= nearDaysTillExpiry)
+            {
+                throw new ArgumentException("CalendarSpread(): expiry arguments must be in ascending order, "
+                    + $"{nameof(nearDaysTillExpiry)}, {nameof(farDaysTillExpiry)}");
+            }
+
+            if (nearDaysTillExpiry < 0)
+            {
+                throw new ArgumentException("CalendarSpread(): near expiry argument must be positive.");
+            }
+            
+            // Select the set strike
+            var strike = AllSymbols.OrderBy(x => Math.Abs(Underlying.Price - x.ID.StrikePrice + strikeFromAtm))
+                .First().ID.StrikePrice;
+            var contracts = AllSymbols.Where(x => x.ID.StrikePrice == strike && x.ID.OptionRight == right);
```

Comment:
Cool!
Couple of thoughts:
- Believe it's a bit sad & fragile having to duplicate the knowledge of each strategy again here, we could have a core implementation which could take a `OptionStrategyDefinition` instance for example and generically handle it there, each of these sugar names could just pass their option strategy definition to the core one and boom, we can simplify many lines and potential issues due to the duplication
- The implementations seem a bit inconsistent on when using `AllSymbols` &/or the helper methods `Where/Contracts/etc` the helper methods are overriding `AllSymbols` & calling to List (sad to do it multiple times etc) -> I think we should avoid them and just use `AllSymbols` but at the **end** of the method call ToList and override it
- Just double checking, I see ur using Abs for date/strike input operations which is strange given the input could be +/- , seems if used in OrderBy it would treat +/- result in the same way which doesn't seem right?

Reply:
> Just double checking, I see ur using Abs for date/strike input operations which is strange given the input could be +/- , seems if used in OrderBy it would treat +/- result in the same way which doesn't seem right?
Yah that was intentional. We look for the best match but not a conditional best match